04/SpeedTest.js

```diff
@@ -1,17 +1,50 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import useRandomItem from './hook';
 
-const SpeedTest = () => {
+const SpeedTest = function () {
     const [word, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JavaScript']);
+    const [counter, setCounter] = useState(0);
+    const [text, setText] = useState(null);
+    const [marks, setMarks] = useState(0);
+    const [words, setWords] = useState(0);
+    const [result, setResult] = useState(0);
+    const intervalRef = useRef(null);
+
+    const stopInterval = () => {
+        clearInterval(intervalRef.current);
+    };
 
     useEffect(() => {
-        regenerateWord();
-    }, []);
+        if (text === word) {
+            setText('');
+            if (words === 10) {
+                setResult(marks / counter);
```

Comment:
Trochę długo trzeba na to czekać ;P
Nie da się od razu wyświetlić odpowiedniej informacji? :P